src/Markdig.Tests/TestContainerBlocks.cs

```diff
@@ -66,5 +66,23 @@ public void CanBeInsertedInto()
             Assert.Throws<ArgumentOutOfRangeException>(() => container.Insert(-1, new ParagraphBlock()));
             Assert.Throws<ArgumentException>(() => container.Insert(0, one)); // one already has a parent
         }
+
+        [Test]
+        public void CanBeSet()
+        {
+            ContainerBlock container = new MockContainerBlock();
+
+            var one = new ParagraphBlock();
+            container.Insert(0, one);
+            Assert.AreEqual(1, container.Count);
+            Assert.AreSame(container[0], one);
```

Comment:
Can you please add asserts around `one`'s parent being set, and the same for the `CanBeInsertedInto` test above